klog.go

```diff
@@ -1370,3 +1371,32 @@ func Exitf(format string, args ...interface{}) {
 	atomic.StoreUint32(&fatalNoStacks, 1)
 	logging.printf(fatalLog, logging.logr, format, args...)
 }
+
+// ObjectRef references a kubernetes object
+type ObjectRef struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+func (ref ObjectRef) String() string {
+	if ref.Namespace != "" {
+		return fmt.Sprintf("%s/%s", ref.Namespace, ref.Name)
+	}
+	return ref.Name
+}
+
+// KObj returns ObjectRef from ObjectMeta
+func KObj(obj metav1.ObjectMeta) ObjectRef {
```

Comment:
The amount of dependencies we need to import just to get this definition is ridiculous.

@deads2k @lavalamp - would you be amenable to extracting metav1 just TYPES into a separate repo?  Or attaching some well-defined getter" methods so consumers like this don't need to hard-couple?

Otherwise, I fear we may have to ask that these helper functions go in a different repo and callers who want them can use them (and everyone else doesn't pay for them).

Reply:
`ObjectMeta` implements `metav1.Object`, mabye define an inteface of  the subset of `metav1.Object` we need here (`GetName`, `GetNamespace`) and use that as type for the `obj` argument?

Reply:
Ahh, I missed those methods.  Yes, that seems to be exactly what I wanted.